feathr-impl/src/main/scala/com/linkedin/feathr/offline/generation/SparkIOUtils.scala

```diff
@@ -74,8 +74,9 @@ object SparkIOUtils {
           val output_format = outputDF.sqlContext.getConf("spark.feathr.outputFormat", "avro")
           // if the output format is set by spark configurations "spark.feathr.outputFormat"
           // we will use that as the job output format; otherwise use avro as default for backward compatibility
-          outputDF.write.mode(SaveMode.Overwrite).format(output_format).save(path)
-          outputDF
+          if(!outputDF.isEmpty) {
```

Comment:
Shouldn't we just write empty dataframe, instead of not producing an output?

Reply:
It throws exception for empty DataFrame.